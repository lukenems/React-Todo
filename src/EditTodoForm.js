import React from 'react';

class EditTodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { editTask: this.props.value, isHidden: this.props.display }
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.updateTodo(this.props.formId, this.state.editTask);

  }

  handleChange(evt) {
    console.log("a change")
    this.setState({ [evt.target.name ]: evt.target.value })
  }
  render() {
    return (
      <form style={{ display: this.state.isHidden }} onSubmit={this.handleSubmit}>
        <label htmlFor="editTask">Edit Task</label>
        <input name="editTask" value={this.state.editTask} onChange={this.handleChange} />
        <button>Save</button>
      </form>
    )
  }
}

export default EditTodoForm