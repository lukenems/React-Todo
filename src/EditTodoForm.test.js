import React from "react";
import { shallow, mount} from "enzyme";
import toJson from "enzyme-to-json";
import EditTodoForm from './EditTodoForm';

it("renders without crashing", function() {
  shallow(<EditTodoForm />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<EditTodoForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})

it("allows for change of new Todo", function() {
  let wrapper = mount(<EditTodoForm value="chore"/>);
  wrapper.instance().handleChange({ target: {
    name: "editTask",
    value: "chores"
  }})

  expect(wrapper.state().editTask).toEqual("chores");
});

it("runs a mocked updateTodo function on submit", function() {
  const submitFn = jest.fn();
  let wrapper = mount(<EditTodoForm updateTodo={submitFn} />);
  const form = wrapper.find("form");

  form.simulate("submit");

  expect(submitFn).toHaveBeenCalled();
});