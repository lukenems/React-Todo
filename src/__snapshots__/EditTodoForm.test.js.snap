// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "display": undefined,
    }
  }
>
  <label
    htmlFor="editTask"
  >
    Edit Task
  </label>
  <input
    id="editTask"
    name="editTask"
    onChange={[Function]}
  />
  <button>
    Save
  </button>
</form>
`;
